packages/react-form/src/hooks/list/hooks/reducer.ts

```diff
@@ -28,6 +29,11 @@ interface AddFieldItemAction<Item> {
   payload: {list: Item[]};
 }
 
+interface MoveFieldItemAction<Item> {
+  type: 'moveFieldItem';
+  payload: {item: Item; oldIndex: number; newIndex: number};
```

Comment:
Why do we need to pass the item? Wouldn't just `oldIndex` and `newIndex` suffice?

I also suggest calling these `fromIndex` and `toIndex`.

Reply:
Yes, it should be sufficient to just pass the indexes and grab the item from the list with `oldIndex`

Reply:
good point, I suppose we can just look up item in `state.list`